includes/classes/class-fixinsecurecontent.php

```diff
@@ -0,0 +1,313 @@
+					'order'          => 'ASC',
+				);
+				$query = new WP_Query( $args );
+
+				if ( $query->have_posts() ) {
+					while ( $query->have_posts() ) {
+						$query->the_post();
+						$this->fix_insecure_content( get_the_ID() );
+					}
+				} else {
+					break;
+				}
+
+				// Set offset for next page.
+				$post_offset += $posts_per_page;
+				$total       += $query->post_count;
+
+				if ( ! defined( 'WP_CLI' ) ) {
+					// when fixing post in a REST request, do it only for the current batch and not until all the posts
+					// have been fixed.
+					break;
+				}
+
+				// Wait for a while before fixing the rest.
+				usleep( 1000 );
+			}
+
+			$this->total_post_count = $total;
+		}
+
+		if ( defined( 'WP_CLI' ) && WP_CLI ) {
+			// translators: Message to show when the fixing of insecure content is completed.
+			$message = PHP_EOL . sprintf( __( 'Total posts checked for insecure URL(s): %s', 'insecure-content-warning' ), $this->total_post_count ) . PHP_EOL;
+			WP_CLI::log( WP_CLI::colorize( "%c{$message}%n " ) );
+			Utils\format_items( 'table', $this->fixed_post_count, array( 'URL(s) fixed summary' ) );
+		} else {
+			if ( null === $this->fixed_post_count ) {
+				return __( 'No post(s) found', 'insecure-content-warning' );
+			}
+
+			$output = '';
+
+			foreach ( $this->warning_count as $warning ) {
+				$warning_message = array_shift( $warning );
+				$output         .= '<span class="warning">' . __( 'Warning', 'insecure-content-warning' ) . '</span>: ' . $warning_message . PHP_EOL;
+			}
+
+			foreach ( $this->fixed_post_count as $fixed_post ) {
+				$output .= array_shift( $fixed_post ) . PHP_EOL;
+			}
+
+			return $output;
+		}
+	}
+
+	/**
+	 * Fix insecure content for given Post ID.
+	 *
+	 * @param int $post_id Post ID.
+	 *
+	 * @return array|string
+	 */
+	protected function fix_insecure_content( $post_id ) {
+		$current_post = get_post( $post_id );
+
+		// Check and make sure post exists.
+		if ( empty( $current_post ) ) {
+			// translators: Message to show when the system is unable to fetch details for the post.
+			$message = sprintf( __( 'Unable to fetch details for post %s', 'insecure-content-warning' ), $post_id );
+			if ( defined( 'WP_CLI' ) && WP_CLI ) {
+				// translators: Message to show when the system is unable to fetch details for the post.
+				WP_CLI::warning( $message );
+			} else {
+				$this->warning_count[] = array( __( 'Warning', 'insecure-content-warning' ) => $message );
+			}
+			return '';
+		}
+
+		// Loop through post content and get HTTPS URLs wherever possible.
+		$post_content = $current_post->post_content;
+		$urls         = $this->parse_content_for_insecure_urls( $post_content );
+		$count        = 0;
+
+		// Check if post content has insecure URLs.
+		if ( empty( $urls ) ) {
+			// translators: Message to show when no insecure content URL found in the post.
+			$message = sprintf( __( 'No insecure content URL found in post %s', 'insecure-content-warning' ), $post_id );
```

Comment:
As above re translator note and use of `%d`

Reply:
Done, thank you.